src/main/java/md/miller1995/Dealership/services/auth/AuthServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package md.miller1995.Dealership.services.auth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import md.miller1995.Dealership.models.dto.auth.UserAuthDTO;
+import md.miller1995.Dealership.models.dto.auth.UserRegisterDTO;
+import md.miller1995.Dealership.models.entities.UserAuthEntity;
+import md.miller1995.Dealership.repositories.UserRepository;
+import md.miller1995.Dealership.securities.JWTUtil;
+import md.miller1995.Dealership.utils.AuthResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly = true)
```

Comment:
If you have @Transactional on the method, it will give us any advantage to use it also in the class?

Reply:
@AlexIliescu11 If I use @Transactional (readOnly=true) on the class, in this case I don't protect my class from some accidental manipulations in the database?               I mean, for the Data Manipulation Language, in this case, it will be necessary to add the additional annotation @Transactional that permit write, on the methods.            Or did I understand something wrong?